### PR TITLE
Update LXQt to version 2.1.0

### LINKED ISSUE #354141: Package request:  lxqt-wayland-session
**Project description**

The LXQt team announces the release of LXQt 2.1.0,  supports 7 Wayland sessions (with Labwc, KWin, Wayfire, Hyprland, Sway, River and Niri), has two Wayland back-ends in lxqt-panel (one for kwin_wayland and the other general)...
https://lxqt-project.org/release/2024/11/05/release-lxqt-2-1-0/

**Metadata**

* homepage URL: https://github.com/lxqt/lxqt-wayland-session
* source URL:  https://github.com/lxqt/lxqt-wayland-session
* license:   GNU Lesser General Public License v2.1
* platforms: unix, linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

@NixOS/lxqt

### PR DESCRIPTION
**Pull request by romildo**
https://lxqt-project.org/release/2024/11/05/release-lxqt-2-1-0/

Closes https://github.com/NixOS/nixpkgs/issues/354141

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/desktops/lxqt/lxqt-wayland-session/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    rev = version;
+    hash = "sha256-5WdfwJ89HWlXL6y9Lpgs7H3mbN/wbf+9VbP9ERPasBM=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    lxqt-build-tools
+    pkg-config
+    qttools
+    wrapQtAppsHook
```
Comment (wineee) [inline pkgs/desktops/lxqt/lxqt-wayland-session/default.nix:34]: lxqt-wayland-session seems to be just a collection of configurations. Does it include any Qt programs?

Comment (romildo) [reply pkgs/desktops/lxqt/lxqt-wayland-session/default.nix:34]:
``` console
$ find /nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/bin
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/bin/startlxqtwayland
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/bin/.startlxqtwayland-wrapped
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/nix-support
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/nix-support/propagated-user-env-packages
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/desk.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/menu.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/themerc
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/iconify.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/shade.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/desk_toggled.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/max_toggled_hover.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/max.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/close.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/max_toggled.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent-dark/openbox-3/shade_toggled.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/desk.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/menu.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/themerc
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/iconify.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/shade.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/desk_toggled.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/max_toggled_hover.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/max.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/close.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/max_toggled.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/themes/Vent/openbox-3/shade_toggled.xbm
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wallpapers
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wallpapers/origami-dark-labwc.png
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/themerc-override
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/rc.xml
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/autostart
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/menu.xml
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/themerc
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/environment
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/labwc/README
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/firstrun
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/firstrun/autostart
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/lxqt-niri.kdl
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/lxqt-river-init
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/lxqt-hyprland.conf
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/lxqt-wayfire.ini
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/wayland/lxqt-sway.config
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/graphics
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/lxqt/graphics/lxqt-labwc.png
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/wayland-sessions
/nix/store/2n94zzjk7gpifn6qpgs8d8w7sns67fym-lxqt-wayland-session-0.1.0/share/wayland-sessions/lxqt-wayland.desktop
```

`bin/.startlxqtwayland-wrapped` is a shell script

Comment (romildo) [reply pkgs/desktops/lxqt/lxqt-wayland-session/default.nix:34]: So no Qt program is included.

Comment (romildo) [reply pkgs/desktops/lxqt/lxqt-wayland-session/default.nix]: I have now removed the dependence on `wrapQtAppsHook`.

Comment (wineee) [reply pkgs/desktops/lxqt/lxqt-wayland-session/default.nix]: There are some hardcoded paths in this project that you need to pay attention to, eg：

https://github.com/lxqt/lxqt-wayland-session/blob/c8cbc996b58bce24c2564b62b99c4f6970106bf3/startlxqtwayland.in#L82

https://github.com/lxqt/lxqt-wayland-session/blob/c8cbc996b58bce24c2564b62b99c4f6970106bf3/configurations/labwc/autostart#L6

Comment (romildo) [reply pkgs/desktops/lxqt/lxqt-wayland-session/default.nix]: Which is more appropriate: replace `/usr/share/X11/xkb/rules/base.lst` by `${xkeyboard-config}/share/X11/xkb/rules/base.lst` or by `/run/current-system/sw/share/X11/xkb/rules/base.lst`?

Comment (wineee) [reply pkgs/desktops/lxqt/lxqt-wayland-session/default.nix]:
```
>>> rg /usr/share/X11/xkb/rules/
pkgs/os-specific/linux/systemd/0008-localectl-use-etc-X11-xkb-for-list-x11.patch
20:-        f = fopen("/usr/share/X11/xkb/rules/base.lst", "re");

pkgs/desktops/lxqt/lxqt-config/default.nix
72:      --replace-fail '/usr/share/X11/xkb/rules/base.lst' \

pkgs/desktops/deepin/core/dde-session-shell/default.nix
49:      --replace "/usr/share/X11/xkb/rules/base.xml" "${xkeyboard_config}/share/X11/xkb/rules/base.xml"

pkgs/applications/misc/synergy/default.nix
60:      --replace "/usr/share/X11/xkb/rules/evdev.xml" "${xkeyboardconfig}/share/X11/xkb/rules/evdev.xml"

pkgs/applications/display-managers/sddm/unwrapped.nix
28:      --replace "/usr/share/X11/xkb/rules/evdev.xml" "${xkeyboardconfig}/share/X11/xkb/rules/evdev.xml"

pkgs/tools/misc/calamares/default.nix
72:    sed -e 's,/usr/share/X11/xkb/rules/base.lst,${xkeyboard_config}/share/X11/xkb/rules/base.lst,' \

```

`${xkeyboard_config}/share/X11/xkb/rules/base.lst ` should be fine

File: pkgs/desktops/lxqt/lxqt-panel/default.nix [hunk 2]
```diff
@@ -79,6 +80,13 @@ stdenv.mkDerivation rec {
     solid
   ];
 
+  patches = [
+    (fetchpatch {
+      url = "https://patch-diff.githubusercontent.com/raw/lxqt/lxqt-panel/pull/2166.patch";
```
Comment (wineee) [inline pkgs/desktops/lxqt/lxqt-panel/default.nix:85]: Upstream made a point release  include this patch
https://github.com/lxqt/lxqt-panel/releases/tag/2.1.1

**Review by wineee — CHANGES_REQUESTED** [review]

File: pkgs/desktops/lxqt/lxqt-wayland-session/default.nix [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    qtsvg
+    xdg-user-dirs
+  ];
+
+  postPatch = ''
+    substituteInPlace startlxqtwayland.in \
+      --replace-fail /usr/share/X11/xkb/rules ${xkeyboard_config}/share/X11/xkb/rules
+
+    for f in configurations/{labwc/autostart,lxqt-hyprland.conf,lxqt-wayfire.ini}; do
+      substituteInPlace $f \
```
Comment (wineee) [inline pkgs/desktops/lxqt/lxqt-wayland-session/default.nix:49]: `substituteInPlace` support  mult-files input

```
substituteInPlace configurations/{labwc/autostart,lxqt-hyprland.conf,lxqt-wayfire.ini} \
    --replace-fail /usr/share/lxqt/wallpapers $out/share/lxqt/wallpapers
```

File: pkgs/desktops/lxqt/lxqt-wayland-session/default.nix [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    lxqt-build-tools
+    pkg-config
+    qttools
+  ];
+
+  buildInputs = [
+    kwindowsystem
+    liblxqt
+    libqtxdg
+    qtsvg
```
Comment (wineee) [inline pkgs/desktops/lxqt/lxqt-wayland-session/default.nix:40]: Does this project require qtsvg?

Comment (romildo) [reply pkgs/desktops/lxqt/lxqt-wayland-session/default.nix:40]: Without it, one gets this error:
```
CMake Error at /nix/store/irwcgpm8csj7br49jih6kdpqkaqya3vf-cmake-3.30.4/share/cmake-3.30/Modules/CMakeFindDependencyMacro.cmake:76 (find_package):
  By not providing "FindQt6Svg.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt6Svg", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt6Svg" with any
  of the following names:

    Qt6SvgConfig.cmake
    qt6svg-config.cmake

  Add the installation prefix of "Qt6Svg" to CMAKE_PREFIX_PATH or set
  "Qt6Svg_DIR" to a directory containing one of the above files.  If "Qt6Svg"
  provides a separate development package or SDK, be sure it has been
  installed.
```

File: pkgs/desktops/lxqt/lxqt-panel/default.nix (hunk 2 above)
Comment (romildo) [reply pkgs/desktops/lxqt/lxqt-panel/default.nix]: Updated.

File: pkgs/desktops/lxqt/lxqt-wayland-session/default.nix (hunk 3 above)
Comment (romildo) [reply pkgs/desktops/lxqt/lxqt-wayland-session/default.nix]: Done.

**Review by jchv — APPROVED** [review]
lxqt-archiver update LGTM.

**Comment by romildo** [comment]
LXQt with X11 is working for me.
I still need to find out how to have LXQt working with wayland.

**Review by wineee — APPROVED** [review]
LGTM,  Wayland session can be completed in a separate PR

**Pull request merged**